### DISCUSSION TITLE
DataTable Filter doesn't work on Widlfly-19

### DISCUSSION THREAD

**Discussion by crocimi** [primefaces/community#490] [PrimeFaces] [answered]
Not sure if this is a Bug, here is what I'm experiencing:
The[ datatable filter showcase example](https://primefaces.org/showcase/ui/data/datatable/filter.xhtml) works if a start it with jetty and primefaces 13.0.3. The filter doens't work if the application ist deployed on Wildfly. 

I'm still on Wildlfy-19.1.0 but I have upgraded the jsf-impl to 2.3.21. I'm not sure if I'm doing everything correct: I took the maven artifact "jakarta.faces" 2.3.21 and manually deleted the "javax" directory (otherwise I had ClassCastExceptions).
All other components (buttons, tables, inputs, dropdown,....) seems to work, the data table filter not. Until Primefaces-12 and jsf-impl-2.3.9.SP08 everything was ok.

Here the relevant test-cases and files:
- standalone project with jetty: primefaces-test-jetty.zip
- primafaces-test.war contained in primefaces-test.zip (to be deployed on Wildfly)
- jsj mojarra 2.3.21
[primefaces-test-jetty.zip](https://github.com/primefaces/community/files/13392670/primefaces-test-jetty.zip)
[primefaces-test.zip](https://github.com/primefaces/community/files/13392676/primefaces-test.zip)
[jakarta.faces-2.3.21.zip](https://github.com/primefaces/community/files/13392680/jakarta.faces-2.3.21.zip)

Any help is great appreciated. Thank you

**Comment by melloware** [comment, marked as answer]
@crocimi have you verified that 2.3.21 Mojarra prints out in your Wildfly logs so you know its actually installed properly?  It sounds like you did it correctly but I just wanted to make sure.

**Comment by crocimi** [comment]
Ah, I think you guessed right, I forgot to update jsf api. Rookie mistake.

Thx a lot!

**Reply by melloware** [reply]
Nice!
